src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,103 @@
+package duke;
+
+import duke.task.*;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A class for storing and retrieving save data for Duke
+ */
+public class Storage {
+    File file;
+
+    /**
+     * Constructs a Storage.
+     *
+     * @param filePath file path of the save data
+     */
+    public Storage(String filePath) {
+        file = new File(filePath);
+    }
+
+    /**
+     * Loads the tasks from the save data.
+     *
+     * @return an ArrayList of tasks retrieved from the save data
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> res = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                res.add(fileLineToTask(sc.nextLine()));
+            }
+            sc.close();
+        } catch (FileNotFoundException e) {
+            try {
+                Files.createDirectories(Paths.get("./data"));
+                File f = new File("data/tasks.txt");
+            } catch (IOException ex) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+        return res;
+    }
+
+    /**
+     * Refreshes the save data.
+     *
+     * @param tasks the TaskList to obtain new data from.
+     */
+    public void refresh(TaskList tasks) {
+        File temp = new File("data/temp.txt");
+        try {
+            FileWriter fw = new FileWriter(temp);
+            fw.write(tasks.toFileFormat());
+            fw.close();
+        } catch (IOException e) {
+            throw new DukeException(e.getMessage());
+        }
+        this.file.delete();
```

Comment:
Maybe you shouldn't use the `this` keyword here since there are no method params shadowing the class variable `file`? [(Reference: Avoid unnecessary use of this with fields)](https://se-education.org/guides/conventions/java/index.html#variables)

```suggestion
       file.delete();
```